### PR TITLE
[FEAT-67]: bookings page completed

### PR DESCRIPTION
**Pull request by bek-shoyatbek**
Bookings page completed. Sorry for big one commit I had an issue with eslint...

[showcase bookings page.webm](https://github.com/user-attachments/assets/b5097297-7a0d-4791-af17-96a5d04899da)

### CODE REVIEW

File: src/modules/home/index.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { SinglePropertyRoute } from './routes/single-property.route';
 import { AuthLayout } from '#/routes/layout-protected.route';
 import { CreateAccommodationRoute } from '#/modules/home/routes/create-acccommodation.route.tsx';
 import { PaymentResult } from './components/payment';
+import { BookingsPageRoute } from './routes/bookings-page.route';
```
Comment (purple-mountain) [inline src/modules/home/index.tsx:7]: use lazyRoute. we already have large bundle size, your current approach will increase it more

File: src/modules/home/components/bookings/booking-card.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      month: 'long',
+      day: 'numeric',
+    });
+  };
+
+  const getStatusColor = (status: BookingStatus) => {
+    if (status === BookingStatus.Pending) return 'warning';
+    if (status === BookingStatus.Confirmed) return 'success';
+    if (status === BookingStatus.CheckedIn || status === BookingStatus.CheckedOut) return 'info';
+    return 'warning';
```
Comment (purple-mountain) [inline src/modules/home/components/bookings/booking-card.tsx:41]: if it is archived, it will also display warning?

File: src/modules/home/api/get-bookings.ts
Comment (purple-mountain) [inline src/modules/home/api/get-bookings.ts:1]: please create a new bookings module and move everything to there. this is not related to home at all

**Review by purple-mountain — CHANGES_REQUESTED** [review]

**Review by purple-mountain — APPROVED** [review]
can you add archived to enum? i think it is better for it to be the same as backend enum

File: src/modules/bookings/components/booking-card.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+type BookingCardProps = {
+  booking: BookingsWithAccommodation;
+};
+
+export function BookingCard({ booking }: BookingCardProps) {
+  const navigate = useNavigate();
+
+  const formatDate = (date: Date) => {
+    return new Date(date).toLocaleDateString('uz-UZ', {
```
Comment (Ismoil-JS) [inline src/modules/bookings/components/booking-card.tsx:30]: you can use dayjs built-in; no need for customization,

File: src/modules/bookings/components/booking-card.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      year: 'numeric',
+      month: 'long',
+      day: 'numeric',
+    });
+  };
+
+  const getStatusColor = (status: BookingStatus) => {
```
Comment (Ismoil-JS) [inline src/modules/bookings/components/booking-card.tsx:37]: you can use switch case intead of else if

File: src/modules/bookings/components/booking-card.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            <Typography variant="h6" component="div" sx={bookingCardStyles.accommodationTitle} onClick={handleAccommodationClick}>
+              <LocationIcon /> {booking.accommodation.name}
+            </Typography>
+          </Box>
+
+          <Box sx={bookingCardStyles.detailsContainer}>
+            <Box sx={bookingCardStyles.dateContainer}>
+              <Box sx={bookingCardStyles.detailBox}>
+                <CalendarIcon color="action" />
+                <Typography variant="body2">Check-in: {formatDate(booking.checkInDate)}</Typography>
```
Comment (Ismoil-JS) [inline src/modules/bookings/components/booking-card.tsx:62]: use dayjs

File: src/modules/bookings/components/bookings.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  if (isLoading) {
+    return (
+      <Box sx={bookingsStyles.loadingContainer}>
+        <CircularProgress />
+      </Box>
+    );
+  }
+
+  if (isError) {
+    toast.error('Error fetching bookings');
```
Comment (Ismoil-JS) [inline src/modules/bookings/components/bookings.tsx:23]: why do you use both toast and alert?

**Review by Ismoil-JS — CHANGES_REQUESTED** [review]

File: src/modules/bookings/components/booking-card.tsx (hunk 4 above)
Comment (farrooh) [reply src/modules/bookings/components/booking-card.tsx:37]: > you can use switch case intead of else if

Or ternary operator

**Review by farrooh — APPROVED** [review]

**Review by Ismoil-JS — APPROVED** [review]
